src/ucp/core/ucp_worker.c

```diff
@@ -498,7 +494,8 @@ ucs_status_t ucp_worker_set_ep_failed(ucp_worker_h worker, ucp_ep_h ucp_ep,
     }
 
     /* In case if this is a local failure we need to notify remote side */
-    if (ucp_ep_is_cm_local_connected(ucp_ep)) {
+    if (ucp_ep_is_cm_local_connected(ucp_ep) &&
+        (status != UCS_ERR_CONNECTION_RESET)) {
```

Comment:
can we drop LOCAL_CONNECTED before calling set_ep_failed, instead of using status as indication?
prefer not to rely on specific status code but on the fact that disconnect callback was called (or connect failed)